### PR TITLE
Support REGEXP_EN_DASH too

### PR DESCRIPTION
**Pull request by GlazerMann**
Yet another dash to recognize in pages

### CODE REVIEW

File: Template.php [hunk 1]
```diff
@@ -558,7 +558,8 @@ public function add_if_new($param_name, $value, $api = NULL) {
       case "page": case "pages":
         if (( $this->blank("pages") && $this->blank("page") && $this->blank("pp")  && $this->blank("p") && $this->blank('at'))
                 || strpos(strtolower($this->get('pages') . $this->get('page')), 'no') !== FALSE
-                || (strpos($value, chr(2013)) || (strpos($value, '-'))
+                || (strpos($value, chr(2013)) || (strpos($value, '-')) || (strpos($value, REGEXP_EN_DASH))
```
Comment (ms609) [inline Template.php]: Is `REGEXP_EN_DASH` compatible with `strpos`?

File: tests/phpunit/TemplateTest.php [hunk 2]
```diff
@@ -864,7 +864,7 @@ public function testJstorSICI() {
    $this->assertEquals('1961', $expanded->get('year'));
    $this->assertEquals('81', $expanded->get('volume'));
    $this->assertEquals('1', $expanded->get('issue'));
-   $this->assertEquals('43', $expanded->get('pages'));
+   $this->assertEquals('43–52', $expanded->get('pages'));
```
Comment (ms609) [inline tests/phpunit/TemplateTest.php]: This test is only designed to check that `43` is extracted from the SICI; if the metadata is found from other sources then the test no longer tests what it is meant to test (and thus needs re-thinking...) – perhaps `process_citation` needs replacing with a function that has a narrower scope.

Comment (GlazerMann) [reply tests/phpunit/TemplateTest.php]: new test added

File: Template.php (hunk 1 above)
Comment (GlazerMann) [reply Template.php]: Shocking yes.  I actually had to test it out.  Good question

**Pull request merged**